Sources/SwiftSyntaxMacros/MacroProtocols/Macro+Format.swift

```diff
@@ -0,0 +1,31 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+/// Describes the mode to use to format the result of an expansion.
+public enum FormatMode {
+  /// Perform a basic format of the expansion. This is primarily for inserting
+  /// whitespace as required (eg. between two keywords), but also adds simple
+  /// newline and indentation.
+  case auto
+
+  /// Disable automatically formatting the expanded macro. Trivia must be
+  /// manually inserted where required (eg. adding spaces between keywords).
+  case disabled
+}
+
+public extension Macro {
+  /// How the resulting expansion should be formatted, `.auto` by default.
+  /// Use `.disabled` for the expansion to be used as is.
+  static var formatMode: FormatMode {
```

Comment:
Don't we need to make this a `Macro` requirement? I thought, without requirements, calls on existential types or archetype types are statically dispatched to the default implementation.

Reply:
Good to know. I'll add a test on the swift side for disabling formatting too.